src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/sunwell_plateau/boss_kiljaeden.cpp

```diff
@@ -119,18 +119,20 @@ enum
     EVENT_DRAGON_ORB            = 9,
 
     // outro
-    SPELL_TELEPORT_VISUAL       = 12980,
+    SPELL_TELEPORT_VISUAL       = 35517,
     SPELL_KALEC_TELEPORT        = 46473,            // teleports and transforms Kalec in human form
     SPELL_ARCANE_PORTAL         = 42047,
     SPELL_CALL_ENTROPIUS        = 46818,
     SPELL_ENTROPIUS_BODY        = 46819,
     SPELL_BLAZE_TO_LIGHT        = 46821,
     SPELL_SUNWELL_IGNITION      = 46822,
+    SPELL_PORTAL_FROM_SHATTRATH = 46801,            // Riftwalker after reached point 1, summon NPC_RIFTWALKER and cast on him this spell.
 
     NPC_INERT_PORTAL            = 26254,
     NPC_CORE_ENTROPIUS          = 26262,
     NPC_SOLDIER                 = 26259,            // summoned in 2 waves before Velen. Should move into 2 circle formations
     NPC_RIFTWALKER              = 26289,
+    NPC_SHATTRATH_PORTAL        = 26251,
```

Comment:
Are you sure this patch is complete?

I don't see any reference to this NPC anywhere

Reply:
Riftwalker on summon, should walk (false) on position, summon npc id 26251 and cast on this npc spell id 46801.
Shattrat_Portal movement on height, meet npc have same entry and despawn. After this actions, summon one npc 26254. Npc 26254 on spawn cast on self 42047 (implemented in SD2) and enjoy!